src/V3AstNodes.cpp

```diff
@@ -1320,6 +1320,90 @@ void AstWhile::addNextStmt(AstNode* newp, AstNode* belowp) {
     }
 }
 
+//======================================================================
+// Debugging
+
+static void checkBitWidth(const AstNode* nodep) VL_MT_STABLE {
+    if (const AstNodeBiop* const biopp = VN_CAST(nodep, NodeBiop)) {
+        // Simple bitwise ops
+        if (VN_IS(biopp, And) || VN_IS(biopp, Or) || VN_IS(biopp, Xor)) {
+            UASSERT_OBJ(biopp->lhsp()->widthMin() == biopp->rhsp()->widthMin(), biopp,
+                        "widthMin mismatch LHS:" << biopp->lhsp()->widthMin()
+                                                 << " RHS:" << biopp->rhsp()->widthMin());
```

Comment:
Print instead dtypep()->dumpSmall as has the widthMin among other info which is useful?

Reply:
checkTreeIter() is marked VT_MT_STABLE, so perhaps I need to also mark AstNodeTYpe::dumpSmall().
I'm not sure how much the mark propagetes, but I will try.